readme.md
A theme for everyone who needs a consistent theme that is available for both GTK and Qt.

![preview](preview.gif)

## Contents

* [Install](#install)
 - [Packages](#packages)
   - [Debian](#debiantesting-unstable)
    - [*buntus](#buntus)
    - [SuSE](#suse-only-tested-on-tumbleweed)
    - [Manjaro/Arch](#manjaroarch-third-party)
 - [Manual](#manual-install)
* [FAQ](#faq)
 - [Qt](#qt)
   - [Plasma issues](#kdeplasma-icon-theme-etc-is-not-applied)
    - [Qt on GTK Desktops](#howto-set-the-qt-style-on-non-qt-desktops)
 - [GTK](#gtk)
   - [GTK on Qt-Desktops](#gtk-on-qt-desktops)
    - [GTK2](#gtk2)
    - [GTK3](#gtk3)
 - [Icons](#icon-theme)
   - [How to change the folder color](#how-to-change-the-folder-color)
    - [Big icons are ugly](#big-icons-are-ugly)
    - [Why no SVG icon theme?](#why-no-svg-icon-theme)
    - [Limitations and Bugs](#limitations-and-bugs)
 - [Xfwm](#xfwm-theme)
 - [Metacity & Co](#metacity-theme)

## Install

### Packages

Official prebuilt packages are available [here](https://github.com/sixsixfive/Hedera/tree/master/pkgs).

#### Debian(testing, unstable)

it is possible to build and install a debian package eg:

```
su -c "apt install --no-install-recommends fakeroot unzip wget debhelper dpkg-dev libfile-fcntllock-perl"
wget https://github.com/sixsixfive/Hedera/archive/master.zip
unzip -q master && cd Hedera-master
fakeroot debian/rules binary
```
#### *buntus

same as debian however use sudo instead of su eg:

```
sudo apt install --no-install-recommends fakeroot unzip wget debhelper dpkg-dev libfile-fcntllock-perl
```

Note: that *buntus usually come with ancient and heavy modified packages, so not all themes might work!

#### SuSE, only tested on Tumbleweed

```
sudo zypper install --no-recommends unzip wget rpm-build findutils sed 
wget https://github.com/sixsixfive/Hedera/archive/master.zip
unzip -q master && cd Hedera-master
sh suse/buildrpm.sh
sudo zypper install --no-recommends hedera-current_tumbleweed.rpm
```

#### Manjaro/Arch (third-party)

* [Packages](http://download.tuxfamily.org/gericom/manjaro) can be found [@gericoms](https://github.com/gericom-hummer) Manjaro Repo.
* It is also available at the [AUR](https://aur.archlinux.org/packages/hedera-icon-theme-git/)

### Manual install

just copy the theme dirs/files to Application theme dirs eg:

```
wget https://github.com/sixsixfive/Hedera/archive/master.zip
7z x master.zip
```

there are 3 folders:

* Plasma -  holds all Plasma specific stuff.
* Other -  holds all non-plasma stuff.

the content of those two folders, usually just needs to be copied to the **$SYS-** or **$LOCALPREFIX**

* Manual - includes stuff that needs some manual work(eg overwriting/backuping files, dnd install)

##### Kinfopages(eg systemsettings classic view, digikam & skrooge startup pages)

```
$SYSPREFIX/share/kf5/infopage/<themefiles>
```

*  KDE4 compat

```
$SYSPREFIX/share/kde4/apps/kdeui/about/<themefiles>
```

## FAQ

### Qt

#### KDE/Plasma: Icon theme etc is not applied

Due Plasma/KDEs caching you might need to re-login!

you can also remove kde cache files under $HOME/.cache to force their recreation eg:

```
find ~/.cache \( -name "*.kcache" -o -name "plasma-*" -o -name "ksycoca5_*" \) -exec rm -f {} \; && kbuildsycoca5
```

#### Howto set the Qt-Style on non Qt-desktops

Install [Qt5ct](https://sourceforge.net/projects/qt5ct/) and add the following to **/etc/X11/Xsession.d/100-qtstyle**

```
export QT_QPA_PLATFORMTHEME=qt5ct
export QT_PLATFORMTHEME=qt5ct
export QT_PLATFORM_PLUGIN=qt5ct
```

or if you have KDE/Plasma besides installed add:

```
export QT_QPA_PLATFORMTHEME=kde
export QT_PLATFORMTHEME=kde
export QT_PLATFORM_PLUGIN=kde
```

to force it over the GNOME integration plugin!

There is also another Tool [QtPalettor](https://github.com/spillerrec/QtPalettor) to set the colors, however haven't tried this yet.

Please note that the Darker subtheme needs Qt5ct due the stylesheet!

### GTK

#### GTK on Qt-Desktops

**Do yourself a favor and don't use the KCM-gtk module to set your GTK themes on KDE, it will mostly break your themes. (eg. theme internal font settings)**

If you have trouble to set the same font size on Qt and GTK make sure that the both toolkits use the same DPI and hinting styles.

There is an [Xsettings-client](https://github.com/KDE/xsettings-kde) available for KDE/Plasma that will help you a lot.You can also use [Xsettingsd](https://github.com/derat/xsettingsd) or the manual way below(NOTE THAT THIS WILL AFFECT ALL DESKTOPS).

#### GTK2!

The GTK2 theme depends on the GTK2 murrine and pixbuf engine it is also recommended to install the mist engine!

create a file ~/.gtkrc-2.0 (globally: /etc/gtk-2.0/gtkrc) with something like:

```
include "$HOME/.local/share/themes/Hedera/gtk-2.0/gtkrc"
gtk-toolbar-style=GTK_TOOLBAR_ICONS
gtk-theme-name="Hedera"
gtk-fallback-icon-theme="hicolor"
gtk-icon-theme-name="Hedera"
gtk-menu-popup-delay=50
gtk-tooltip-timeout=1000
gtk-enable-mnemonics=1
gtk-auto-mnemonics=1
gtk-alternative-button-order=1
gtk-recent-files-enabled=0
gtk-enable-animations=1
gtk-menu-images=1
gtk-error-bell=0
gtk-button-images=1
gtk-cursor-theme-name="breeze_cursors"
gtk-menu-images=1
gtk-xft-dpi=96
gtk-font-name="Noto Sans Regular 10"
```

* See also: [GTK2 Settings](https://developer.gnome.org/gtk2/stable/GtkSettings.html#GtkSettings.properties)

#### GTK3

The GTK3 theme recommends [GTK3-NOCSD](https://github.com/PCMan/gtk3-nocsd) and to disable the GTK3 overlay scrollbars (export GTK_OVERLAY_SCROLLING=0)!

create a file ~/.config/gtk-3.0/settings.ini (globally: /etc/gtk-3.0/settings.ini) with something like:

```
[Settings]
gtk-theme-name=Hedera
gtk-icon-theme-name=Hedera
gtk-shell-shows-app-menu=0
gtk-shell-shows-menubar=0
gtk-dialogs-use-header=0
gtk-menu-popup-delay=50
gtk-tooltip-timeout=1000
gtk-enable-mnemonics=1
gtk-auto-mnemonics=1
gtk-alternative-button-order=1
gtk-recent-files-enabled=0
gtk-dialogs-use-header=0
gtk-enable-animations=1
gtk-menu-images=1
gtk-enable-accels=1
gtk-decoration-layout=":menu"
gtk-button-images=1
gtk-fallback-icon-theme=hicolor
gtk-cursor-theme-name=breeze_cursors
gtk-menu-images=1
gtk-xft-dpi=96
gtk-font-name="Noto Sans Regular 10"
``` 

* See also: [GTK3 Settings](https://developer.gnome.org/gtk3/stable/GtkSettings.html#GtkSettings.properties)

### Icon theme

#### How to change the folder color

You can change the color of the default folders with the config script eg:

    su -c "/usr/share/icons/Hedera/config.sh"

#### Big icons are ugly

The theme is made for 48px and I wouldn't recommend to go highter than 96px however there are higher sizes included since KDE requires them and they can be used on HiDPI desktops.

#### Why no SVG icon theme

* poor support by both GTK(has issues with masks) and Qt(supports only SVG tiny 1.1).
* PNGs require more space but the performance is alot better since everything is already prerendered

#### Limitations and Bugs

* some smaller icons are fuzzy cause they are scaled down from other sizes(sry, but I don't have plenty of time but someday they will be complete I guess ;)
* some icons are still missing (thats an endless task) 
* Xfce misses some features, since exo can't follow symlinks[10126](https://bugzilla.xfce.org/show_bug.cgi?id=10126)
* Due Qt-bug [33123](https://bugreports.qt.io/browse/QTBUG-33123) & [43620](https://bugreports.qt.io/browse/QTBUG-43620) all extra folders are ignored (Workaround HederaQt available)

### Xfwm theme

The most options are hardcoded in this theme you can only change them by editing the themerc

### Metacity theme

Like all other WM-themes this one was created for the use with a composite extension(shadows). Also note that there is no way to change the radius for the borders in metacity so this theme looks different than other themes.


debian/changelog
hedera (5.9.7-1) unstable; urgency=low

  * many updates to the darker theme

 -- SSF <https://github.com/sixsixfive>  Thu, 09 Mar 2017 00:10:22 +0100

hedera (5.9.6-1) unstable; urgency=low

  * removal of the classic theme
  * darker theme was added(depends on qt5ct!)
  * start adding symbolic icons(manly for use on the contrast panel & headerbars
  * better GTK3 legacy window support
  * better xfce GTK3 components
  * some cleaning

 -- SSF <https://github.com/sixsixfive>  Fri, 03 Mar 2017 18:12:23 +0100

hedera (5.9.5-1) unstable; urgency=low

  * contrast gtk3 theme has now a dark csd headerbar needs adwaita due symbolic icons
  * chrome has now a dark frame
  * some cleaning

 -- SSF <https://github.com/sixsixfive>  Sat, 25 Feb 2017 21:56:43 +0100

hedera (5.9.4-1) unstable; urgency=low

  * matching onboard theme
  * better small icons
  * some GTK3 changes(hopefully improvements)
  * added contrast theme
  * the debian postinstall will now ask to run the icon config script

 -- SSF <https://github.com/sixsixfive>  Wed, 22 Feb 2017 20:02:18 +0100

hedera (5.9.3-1) unstable; urgency=low

  * reworked the console colorshemes(whiptail should now look better)
  * some icon fixes
  * some GTK2 fixes
  * some GTK3 fixes

 -- SSF <https://github.com/sixsixfive>  Wed, 15 Feb 2017 23:48:38 +0100

hedera (5.9.2-1) unstable; urgency=low

  * better xfdashbord theme
  * some tiny gtk2 fixes
  * some icon fixes
  * xfce4-terminal fixes
  * terminator fixes
    
 -- SSF <https://github.com/sixsixfive>  Sun, 12 Feb 2017 12:27:07 +0100

hedera (5.9.1-1) unstable; urgency=low

  * more compatible with the GUI theme
  * some plasma updates
  * xfdashboard theme
    
 -- SSF <https://github.com/sixsixfive>  Thu, 10 Feb 2017 21:01:09 +0100

hedera (5.9-1) unstable; urgency=low

  * use downscaling in the icon theme(reduces size by ~70%)
  * some tiny fixes
    
 -- SSF <https://github.com/sixsixfive>  Wed, 01 Feb 2017 05:59:21 +0100

hedera (5.8.005-1) unstable; urgency=low

  * some gtk fixes
  * updated the icons
  * non-colored new wm theme
  * make the new theme default
  * impove CSD support so(window with CSDs now look almost like the new wm theme, so no hard dep on gtk3-nocsd anymore)
    
 -- SSF <https://github.com/sixsixfive>  Sat, 28 Jan 2017 21:59:30 +0100

hedera (5.8.004-1) unstable; urgency=low

  * many gtk3 scrollbar fixes
    
 -- SSF <https://github.com/sixsixfive>  Sat, 21 Jan 2017 15:12:16 +0100

hedera (5.8.003-1) unstable; urgency=low

  * more unity fixes
  * some more gtk3 fixes
  
 -- SSF <https://github.com/sixsixfive>  Tue, 18 Jan 2017 16:26:43 +0100

hedera (5.8.002-1) unstable; urgency=low

  * some unity fixes
  * some gnome app fixes
  * updated the icons

 -- SSF <https://github.com/sixsixfive>  Tue, 17 Jan 2017 19:26:43 +0100

hedera (5.8.001-1) unstable; urgency=low

  * more fixes
  * new folder icons
  * its now possible to change the color of folders
  
 -- SSF <https://github.com/sixsixfive>  Sat, 14 Jan 2017 17:52:37 +0100

hedera (5.8-1) unstable; urgency=low

  * more fixes - again many gtk fixes
  * using Plasma/breeze version numbers
  
 -- SSF <https://github.com/sixsixfive>  Thu, 12 Jan 2017 17:29:12 +0100

hedera (1.7.3-1) unstable; urgency=low

  * some fixes
  * better hilight color that has better contrast with text
  * all wm themes reworked
  * kde like scrollbar
  * symlinking all the wm themes
  
 -- SSF <https://github.com/sixsixfive>  Mon, 09 Jan 2017 00:04:34 +0100


hedera (1.7.2-1) unstable; urgency=low

  * some very simple fixes
  * gtk3 works now with firefox >=50.x
  
 -- SSF <https://github.com/sixsixfive>  Sat, 07 Jan 2017 01:08:18 +0100

hedera (1.7.1-1) unstable; urgency=low

  * many fixes for the icon theme
  * switching isnt supported anymore since the qt theme is now symlinked as a seperate theme

 -- SSF <https://github.com/sixsixfive>  Sun, 01 Jan 2017 17:13:38 +0100


hedera (1.7-1) unstable; urgency=low

  * a huge update for GTK3.

 -- SSF <https://github.com/sixsixfive>  Thu, 31 Dec 2016 20:53:33 +0100

hedera (1.6.5-1) unstable; urgency=low

  * updated the icons
  * reworked the shadows on the xfwm themes
  * added chrome scrollbars
  * a lot of tiny, tiny fixes
  * provide only one GTK3 release(both were the same)

 -- SSF <https://github.com/sixsixfive>  Thu, 29 Dec 2016 21:53:33 +0100

hedera (1.6.2-1) unstable; urgency=low

  * more small icons
  * more distro support
  * the deb can now provide the adwaita and gnome icon themes
  * the gtk3 theme should now be compatible with the lightdm-gtk-greeter/light-locker

 -- SSF <https://github.com/sixsixfive>  Tue, 27 Dec 2016 13:13:46 +0100

hedera (1.6.1-1) unstable; urgency=low

  * some minor gtk3 fixes
  * added wm themes with less and without borders
  * added a plank theme

 -- SSF <https://github.com/sixsixfive>  Mon, 26 Dec 2016 15:05:09 +0100

hedera (1.6-1) unstable; urgency=low

  * some GTK2 and 3 fixes
  * xfwm buttons on right(everybody knows how to make them left!)
  * xfce-panel theme
  * add support for windowck and unity
  * xfwm theme will qllow to hide its borders on maximize so it can use the windowck plugin

 -- SSF <https://github.com/sixsixfive>  Sun, 25 Dec 2016 13:58:52 +0100

hedera (1.5.0-1) unstable; urgency=low

  * some tiny fixes
  * added the xfwm and metacity theme

 -- SSF <https://github.com/sixsixfive>  Sat, 24 Dec 2016 11:10:25 +0100

hedera (1.4.9-1) unstable; urgency=low

  * many fixes
  * added balou & xfce notifyd theme
  * many fixes for xfce and mate
  
 -- SSF <https://github.com/sixsixfive>  Wed, 22 Dec 2016 04:42:11 +0100

hedera (1.4.6-1) unstable; urgency=low

  * some GTK3 fixes
  * some GTK desktop fixes
  
 -- SSF <https://github.com/sixsixfive>  Wed, 21 Dec 2016 04:42:11 +0100

hedera (1.4.5-1) unstable; urgency=low

  * some tiny bug fixes here
  * the lxqt theme bas been reworked
  
 -- SSF <https://github.com/sixsixfive>  Sun, 18 Dec 2016 00:33:12 +0100

hedera (1.4-1) unstable; urgency=low


Other/icons/Hedera/config.sh
#!/bin/sh
_recolorfolders="action-albumfolder-importdir2.png
add-folder-to-archive.png
albumfolder-importdir.png
albumfolder-importimages.png
albumfolder-new.png
applications-accessories.png
applications-development-translation.png
applications-development-web.png
applications-development.png
applications-education-language.png
applications-education-mathematics.png
applications-education-miscellaneous.png
applications-education-preschool.png
applications-education-school.png
applications-education-science.png
applications-education-university.png
applications-education.png
applications-engineering.png
applications-games.png
applications-graphics.png
applications-internet.png
applications-multimedia.png
applications-office.png
applications-other.png
applications-science.png
applications-system.png
applications-toys.png
applications-utilities.png
archive-insert-directory.png
bookmark_folder.png
cxmenu-cxoffice-0-cxmenu.png
cxmenu.png
dialog-open.png
directory-closed.png
directory.png
document-open-data.png
document-open-folder.png
document-open-recent.png
document-open-remote.png
document-open.png
e-module-gadman.png
favorites.png
fileopen.png
folder-activities.png
folder-add.png
folder-bookmark.png
folder-cloud.png
folder-development.png
folder-documents.png
folder-download.png
folder-downloads.png
folder-dropbox.png
folder-drag-accept.png
folder-favorites.png
folder-gdrive.png
folder-html.png
folder-image-people.png
folder-image.png
folder-images.png
folder-important.png
folder-mail.png
folder-music.png
folder-network.png
folder-new.png
folder-onedrive.png
folder-open.png
folder-owncloud.png
folder-picture.png
folder-pictures.png
folder-print.png
folder-public.png
folder-publicshare.png
folder-remote.png
folder-script.png
folder-sound.png
folder-tar.png
folder-templates.png
folder-text.png
folder-video.png
folder-videos.png
folder-videos.png
folder.png
folder_color_downloads.png
folder_favorite.png
folder_html.png
folder_important.png
folder_new.png
folder_open.png
folder_sound.png
folder_txt.png
folder_video.png
folder_videos.png
folder_wordprocessing.png
folder_home.png
folder_home2.png
folder-home2.png
gfpm-instfromfile.png
gmpc-database.png
gnome-applications.png
gnome-devel.png
gnome-fs-desktop.png
gnome-fs-home.png
gnome-fs-directory.png
gnome-fs-ftp.png
gnome-fs-network.png
gnome-fs-nfs.png
gnome-fs-share.png
gnome-fs-smb.png
gnome-fs-ssh.png
gnome-globe.png
gnome-graphics.png
gnome-joystick.png
gnome-mime-x-directory-smb-share.png
gnome-mime-x-directory-smb-workgroup.png
gnome-multimedia.png
gnome-other.png
gnome-system.png
gnome-util.png
gtk-directory.png
gtk-network.png
gtk-open.png
inode-directory.png
insync-folder.png
knetattach.png
library-music.png
mate-panel-drawer.png
modules-system.png
modules-utils.png
network-server.png
network-workgroup.png
network.png
network_local.png
new-folder.png
open.png
org.kde.plasma.folder.png
package_development.png
package_games.png
package_graphics.png
package_multimedia.png
package_network.png
package_office.png
package_system.png
package_utilities.png
package_wordprocessing.png
plasmaapplet-shelf.png
preferences-directories.png
preferences-other.png
project-open.png
redhat-home.png
redhat-accessories.png
redhat-games.png
redhat-graphics.png
redhat-internet.png
redhat-office.png
redhat-programming.png
redhat-sound_video.png
redhat-system.png
redhat-system_tools.png
sidebar-places.png
stock_folder.png
stock_internet.png
stock_new-dir.png
stock_open.png
tag-folder.png
user-bookmarks.png
user-home.png
user-desktop.png
x-clementine-albums.png
xfce-devel.png
xfce-games.png
xfce-graphics.png
xfce-internet.png
xfce-multimedia.png
xfce-office.png
xfce-sound.png
xfce-system.png
xfce-unknown.png
xfce-utils.png
yast-nfs.png
yast-samba-client.png"

set -e
if [ ! -t 0 ]; then
	if type x-terminal-emulator &>/dev/null; then
		x-terminal-emulator -e "$0"
		exit 0
	else
		exit 1
	fi
fi
_basedir="$(dirname "$(readlink -f "${0}")")"
cd $_basedir
if [ ! -f ./icon-theme.cache ];then
	printf "You have to build the theme fist!\n"
	sleep 5
	exit 1
fi

auto_distroicon() {
	if [ -f /etc/debian_version ];then
		if dpkg --get-selections|grep siduction ;then
			_distributor="siduction"
		elif dpkg --get-selections|grep devuan ;then
			_distributor="devuan"
		elif dpkg --get-selections|grep trios ;then
			_distributor="trios"
		elif dpkg --get-selections|grep linuxmint ;then
			_distributor="linuxmint"
		elif dpkg --get-selections|grep ubuntu ;then
			_distributor="ubuntu"
		elif dpkg --get-selections|grep kfreebsd ;then
			_distributor="freebsd"
		else
			_distributor="debian"
		fi
	fi
	if [ -f /etc/SUSE-brand ];then
		_distributor="suse"
	fi
	if [ -f /etc/devuan_version ];then
		_distributor="devuan"
	fi
	if [ -f /etc/manjaro-release ];then
		_distributor="manjaro"
	fi
	if [ -f /etc/gentoo-release ];then
		_distributor="gentoo"
	fi
	if [ -f /etc/alpine-release ];then
		_distributor="alpine"
	fi
	if [ -f /etc/slackware-version ];then
		_distributor="slackware"
	fi
	if [ -f /etc/salix-update-notifier.conf ];then
		_distributor="salix"
	fi
	if [ -f /etc/GoboLinuxVersion ];then
		_distributor="gobo"
	fi
	if [ -f /etc/freebsd-update.conf ];then
		_distributor="freebsd"
	fi
#systemd
	if [ -f /etc/os-release ]; then
		if [ $(cat /etc/os-release|grep "^NAME=Puppy$") ];then
			_distributor="puppy"
		elif [ $(cat /etc/os-release|grep "^ID=chakra$") ];then
			_distributor="chakra"
		elif [ $(cat /etc/os-release|grep "^ID=ubuntu$") ];then
			_distributor="ubuntu"
		elif [ $(cat /etc/os-release|grep "^ID=neon$") ];then
			_distributor="ubuntu"
		elif [ $(cat /etc/os-release|grep "^ID=linuxmint$") ];then
			_distributor="ubuntu"
		fi
	fi
	if [ -f "${_basedir}"/48/logos/distributor-$_distributor.png ];then
		for _dir in $(echo $(find "${_basedir}" -maxdepth 1 -mindepth 1 -type d -not -name symbolic));do
			cd $_dir
			cp -fv "$_dir"/logos/distributor-$_distributor.png "$_dir"/logos/emblem-distributor.png
			cd $_basedir
		done
	fi
}

custom_distroicon() {
	printf "The following icons are currently available:\n"
	find "${_basedir}"/48/logos -name "distributor-*.png"|sed 's/^.*-//g;s/.png//g'
	printf "\nPlease enter the name of the icon(eg: kde for distributor-kde.png)\n\n"
	read _customiconname
	if [ ! -f "${_basedir}"/48/logos/distributor-$_customiconname.png ];then
		printf "\ndistributor-$_customiconname.png does not exist - Aborting!\n"
		exit 1
	else
		_distributor="$_customiconname"
	fi
	for _dir in $(echo $(find "${_basedir}" -maxdepth 1 -mindepth 1 -type d -not -name symbolic));do
		cd $_dir
		cp -fv "$_dir"/logos/distributor-$_distributor.png "$_dir"/logos/emblem-distributor.png
		cd $_basedir
	done
}

reset_distroicon() {
	for _dir in $(echo $(find "${_basedir}" -maxdepth 1 -mindepth 1 -type d -not -name symbolic));do
		cd $_dir
		cp -fv "$_dir"/logos/emblem-ivy.png "$_dir"/logos/emblem-distributor.png
		cd $_basedir
	done
}

toggleqtworkaround() {
	#if [ -f index.theme.xdg ];then
		#printf "disabling Qt-workaround\n"
		#mv -v index.theme index.theme.qt
		#mv -v index.theme.xdg index.theme
	#elif [ -f index.theme.qt ];then
		#printf "enabling Qt-workaround\n"
		#mv -v index.theme index.theme.xdg
		#mv -v index.theme.qt index.theme
	#fi
	printf "removed!\n"
}

createsymbolicsymlinks() {
	printf "removed!\n"
	#cd $_basedir
	#unset "_tmpicons"
	#if [ -d /usr/share/icons ];then
		#_prefix=/usr
	#elif [ -d $(getconf PATH| sed 's/\/bin//g;s/://g')/share/icons ];then
		#_prefix=$(getconf PATH| sed 's/\/bin//g;s/://g')
	#else
		#printf "couldn't find prefix"
	#fi
	#if [ -d $_prefix/share/icons/[Gg]nome ]; then
		#_tmpicons="$_tmpicons\n$(find $_prefix/share/icons/[Gg]nome -mindepth 1 -type f -iname "*symbolic*" -printf "%f\n"|sed 's/.svg/.png/g')"
	#else
		#printf "The Gnome theme isn't installed you will probably miss some icons"
		#sleep 3
	#fi
	#if [ -d $_prefix/share/icons/[Aa]dwaita ]; then
		#_tmpicons="$_tmpicons\n$(find $_prefix/share/icons/[Aa]dwaita -mindepth 1 -type f -iname "*symbolic*" -printf "%f\n"|sed 's/.svg/.png/g')"
	#else
		#printf "The Adwaita theme isn't installed you will probably miss some icons"
		#sleep 3
	#fi
	#if [ -d $_prefix/share/icons/[Hh]icolor ]; then
		#_tmpicons="$_tmpicons\n$(find $_prefix/share/icons/[Hh]icolor -mindepth 1 -type f -iname "*symbolic*" -printf "%f\n"|sed 's/.svg/.png/g')"
	#fi
	#printf "$_tmpicons" > symboliclinks
	#sort symboliclinks|uniq > icons
	#cd 16
	#if [ ! -f $_basedir/pre_symbolicicons ];then
		#find misc-symbolic >$_basedir/pre_symbolicicons
	#fi
	#if [ ! -d misc-symbolic ];then
		#mkdir misc-symbolic
	#fi
	#for _icon in $(cat ../icons); do
		#if [ -f "misc/$(echo "$_icon"|sed 's/-symbolic.symbolic.png/.png/')" ];then
			#if [ ! -L misc-symbolic/$_icon ]; then
				#ln -sv ../misc/$(echo $_icon|sed 's/-symbolic.symbolic.png/.png/') misc-symbolic/$_icon
			#fi
		#fi
		#if [ -f "misc/$(echo "$_icon"|sed 's/-symbolic.png/.png/')" ];then
			#if [ ! -L misc-symbolic/$_icon ]; then
				#ln -sv ../misc/$(echo $_icon|sed 's/-symbolic.png/.png/') misc-symbolic/$_icon
			#fi
		#fi
	#done
	#cd $_basedir
	#rm symboliclinks icons
	#for _dir in $(echo $(find -maxdepth 1 -mindepth 1 -type d)|sed 's#./16##');do
			#cp -r 16/misc-symbolic $_dir
	#done
}

removesymbolicsymlinks() {
	printf "removed!\n"
	#cd $_basedir
	#cd 16
	#find misc-symbolic >$_basedir/post_symbolicicons
	#_createdicons=$(diff $_basedir/pre_symbolicicons $_basedir/post_symbolicicons|grep "[<>]"|sort|sed 's/^[<>] //'|uniq -u)
	#cd $_basedir
	#for _dir in $(echo $(find -maxdepth 1 -mindepth 1 -type d)|sed 's#./16##');do
		#cd $_dir
		#for _createdicon in $(echo $_createdicons); do
			#if [ -f $_createdicon ];then
				#rm -vf $_createdicon
			#fi
		#done
		#cd $_basedir
	#done
	#rm -f $_basedir/pre_symbolicicons $_basedir/post_symbolicicons
}

recolorfolders() {
	cd $_basedir
	printf "Please enter a color name! \n(blue, pink, violet, green, red, yellow, grey, orange, white, ecru(default), brown, cyan)\n\n"
	read _newcolor
	if [ ! -f "${_basedir}"/48/folders/folder-$_newcolor.png ];then
		printf "\n$_newcolor is not a valid color! - Aborting!\n"
		exit 1
	else
		_oldcolor=$(readlink "${_basedir}"/48/misc-filesystems/inode-directory.png|sed 's/^.*-//g;s/\..*$//g')
		for _recolorfolder in $(echo $_recolorfolders); do
			for _reallink in $(echo $(find ${_basedir} -name "$_recolorfolder")); do
				ln -sf $(readlink $_reallink|sed 's/'$_oldcolor'/'$_newcolor'/g') $_reallink
			done
		done
	fi
}


rebuildgtkiconcache() {
	if type gtk-update-icon-cache &>/dev/null; then
        rm -rf $_basedir/icon-theme.cache
		gtk-update-icon-cache $_basedir
	elif type gtk-update-icon-cache-3.0 &>/dev/null; then
        rm -rf $_basedir/icon-theme.cache
		gtk-update-icon-cache-3.0 $_basedir
	fi
	if [ ! -f $_basedir/icon-theme.cache ]; then
		printf "\nIcon cache creation failed!, is gtk-update-icon-cache installed?\n\n"
	fi
}

case "$1" in
    -i)
		auto_distroicon
		rebuildgtkiconcache
		exit 0
    ;;
esac

while [ 1 ];do
	clear
	printf "\nWhat would you like to do?:\n
#1: Try to automatically set the distributor icon
#2: Set a custom distributor icon
#3: Reset distributor icon
#4: Change the current folder color
#5: Rebuild GTK+ icon cache
#6: Exit this script\n\n"
	printf "Make your choice: [1,2,3,4,5,6,7,8,9]"
	read _choice
	case $_choice in
		1)
			clear
			auto_distroicon
			rebuildgtkiconcache
			sleep 5
		;;
		2)
			clear
			custom_distroicon
			rebuildgtkiconcache
			sleep 5
		;;
		3)
			clear
			reset_distroicon
			rebuildgtkiconcache
			sleep 5
		;;
		4)
			clear
			recolorfolders
			rebuildgtkiconcache
			sleep 5
		;;
		5)
			clear
			rebuildgtkiconcache
			sleep 5
		;;
		6)
			printf "\nbye\n"
			sleep 2
			exit 0
			break
		;;
		*)
			printf '\nnothing more here\n'
		;;
	esac
done
